codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -195,14 +195,26 @@ private FlowState connectionState() {
     }
 
     private static FlowState state(Http2Stream stream) {
-        checkNotNull(stream, "stream");
-        return stream.getProperty(FlowState.class);
+        if (stream instanceof FastPropertyAccess) {
```

Comment:
I the approach I described above works nicely.  I will submit a PR for this approach on Friday and we can evaluate it then.